src/widgets/tree/mod.rs

```diff
@@ -0,0 +1,493 @@
+//! A tree widget.
+use std::collections::HashSet;
+
+use unicode_width::UnicodeWidthStr;
+
+use crate::{prelude::*, widgets::*};
```

Comment:
Why are these star imports needed? They don't seem specific and are therefor a bit shady to me. Explicit imports seem like a better idea to me and personally I would just import stuff like Text, Block and so on explicitly.

Guess the prelude in itself is highly based on preference and personally I try to follow all the pedantic clippy lints including [wildcard_imports](https://rust-lang.github.io/rust-clippy/master/index.html#/wildcard_imports) (which allows prelude::* by default but I don't really see the point in less explicit imports for a specific name like prelude)